### PR TITLE
logger move non-logged subscriptions to uORB::Subscription

### PR DESCRIPTION
**Pull request by dagar**
Smaller pieces broken out of https://github.com/PX4/Firmware/pull/12123.

### CODE REVIEW

File: src/modules/logger/log_writer_mavlink.cpp [hunk 1]
```diff
@@ -62,20 +60,11 @@ LogWriterMavlink::~LogWriterMavlink()
...
 
 void LogWriterMavlink::start_log()
 {
-	if (_ulog_stream_pub == nullptr) {
-		memset(&_ulog_stream_data, 0, sizeof(_ulog_stream_data));
-		// advertise before we subscribe: this is a trick to reduce the number of needed file descriptors
-		// (the advertise temporarily opens a file descriptor)
-		_ulog_stream_pub = orb_advertise_queue(ORB_ID(ulog_stream), &_ulog_stream_data, _queue_size);
-	}
+	// make sure we don't get any stale ack's by doing an orb_copy
```
Comment (bkueng) [inline src/modules/logger/log_writer_mavlink.cpp]: This is incorrect here.

File: src/modules/logger/logger.cpp [hunk 2]
```diff
@@ -898,11 +893,6 @@ void Logger::run()
 		}
 	}
 
-	int vehicle_status_sub = orb_subscribe(ORB_ID(vehicle_status));
-	uORB::SubscriptionData<parameter_update_s> parameter_update_sub(ORB_ID(parameter_update));
```
Comment (bkueng) [inline src/modules/logger/logger.cpp]: Can you keep that here? It does not need to be a class attribute.

File: src/modules/logger/log_writer_mavlink.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/logger/log_writer_mavlink.cpp]: Fixed.

File: src/modules/logger/logger.cpp (hunk 2 above)
Comment (dagar) [reply src/modules/logger/logger.cpp]: Moved back.

**Pull request merged**